x-pack/solutions/security/plugins/security_solution/server/lib/detection_engine/rule_types/esql/esql.test.ts

```diff
@@ -0,0 +1,128 @@
+import { alertsMock } from '@kbn/alerting-plugin/server/mocks';
+import { getExceptionListItemSchemaMock } from '@kbn/lists-plugin/common/schemas/response/exception_list_item_schema.mock';
+import type { ExperimentalFeatures } from '../../../../../common';
+import { getIndexVersion } from '../../routes/index/get_index_version';
+import { SIGNALS_TEMPLATE_VERSION } from '../../routes/index/get_signals_template';
+import type { EsqlRuleParams } from '../../rule_schema';
+import { getCompleteRuleMock, getEsqlRuleParams } from '../../rule_schema/mocks';
+import { ruleExecutionLogMock } from '../../rule_monitoring/mocks';
+import { esqlExecutor } from './esql';
+import { getDataTierFilter } from '../utils/get_data_tier_filter';
+
+jest.mock('../../routes/index/get_index_version');
+jest.mock('../utils/get_data_tier_filter', () => ({ getDataTierFilter: jest.fn() }));
+
+const getDataTierFilterMock = getDataTierFilter as jest.Mock;
+
+describe('esqlExecutor', () => {
+  const version = '9.1.0';
+  const ruleExecutionLogger = ruleExecutionLogMock.forExecutors.create();
+  let alertServices: RuleExecutorServicesMock;
+  (getIndexVersion as jest.Mock).mockReturnValue(SIGNALS_TEMPLATE_VERSION);
+  const params = getEsqlRuleParams();
+  const esqlCompleteRule = getCompleteRuleMock<EsqlRuleParams>(params);
+  const tuple = {
+    from: dateMath.parse(params.from)!,
+    to: dateMath.parse(params.to)!,
+    maxSignals: params.maxSignals,
+  };
+  const mockExperimentalFeatures = {} as ExperimentalFeatures;
+  const mockScheduleNotificationResponseActionsService = jest.fn();
+  const SPACE_ID = 'space';
+  const PUBLIC_BASE_URL = 'http://testkibanabaseurl.com';
+
+  let mockedArguments: Parameters<typeof esqlExecutor>[0];
+
+  beforeEach(() => {
+    jest.clearAllMocks();
+    alertServices = alertsMock.createRuleExecutorServices();
+    getDataTierFilterMock.mockResolvedValue([]);
+
+    mockedArguments = {
+      runOpts: {
+        completeRule: esqlCompleteRule,
+        tuple,
+        ruleExecutionLogger,
+        bulkCreate: jest.fn(),
+        mergeStrategy: 'allFields',
+        primaryTimestamp: '@timestamp',
+        alertWithSuppression: jest.fn(),
+        unprocessedExceptions: [getExceptionListItemSchemaMock()],
+        publicBaseUrl: PUBLIC_BASE_URL,
+      },
+      services: alertServices,
+      version,
+      licensing: {},
+      spaceId: SPACE_ID,
+      experimentalFeatures: mockExperimentalFeatures,
+      scheduleNotificationResponseActionsService: mockScheduleNotificationResponseActionsService,
+    } as unknown as Parameters<typeof esqlExecutor>[0];
+  });
+
+  describe('eqlExecutor', () => {
+    describe('errors', () => {
+      it('should return result with user error equal true when request fails with data verification exception', async () => {
+        alertServices.scopedClusterClient.asCurrentUser.transport.request.mockRejectedValue(
+          new KbnServerError(
```

Comment:
I realize the work involved in capturing an accurate representation of an error like this. It would be nice to get dividends from that work!

Can we capture these errors in a separate file, e.g. `mock_alerting_error_responses.ts` or something like that? That would allow us to name and qualify them, e.g. "This represents an elasticsearch error response caused by a data verification exception" which would likely also help with understanding/reusability.